src/Features/CSharp/Portable/AddParameter/CSharpAddParameterCodeFixProvider.cs

```diff
@@ -1,17 +1,18 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
-using System;
 using System.Collections.Immutable;
 using System.Composition;
 using Microsoft.CodeAnalysis.AddParameter;
 using Microsoft.CodeAnalysis.CodeFixes;
 using Microsoft.CodeAnalysis.CSharp.GenerateConstructor;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
 
 namespace Microsoft.CodeAnalysis.CSharp.AddParameter
 {
     [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.AddParameter), Shared]
     [ExtensionOrder(Before = PredefinedCodeFixProviderNames.GenerateConstructor)]
+    [ExtensionOrder(After = PredefinedCodeFixProviderNames.GenerateMethod)]
```

Comment:
The idea is to place this fix below *Generate method*. I tried several combinations (with the help of @CyrusNajmabadi on gitter) but I wasn't able to get ordering right. The *Add parameter* fix stayed always above *Generate method*. I also tried deleting the MEF cache. I need help to get this working.

Reply:
Note: i don't care about this as much anymore.  This was important if there might be a dozen items added to teh smart tag.  If we're capping at essentially '4', this is a lot less important.  

However, if we do want to move down, i'm fine with it being below GenConstructor as well.  Note:

> The problem is that "GenerateConstructor" and "GenerateMethod" belong to the same codefixer

This could always be split out.  Abstract base class with all the logic.  Two separate fixers, just so we could control ordering.

Reply:
Given that GenerateMethod is a non-breaking change (code compiles after it's applied) and AddParameter isn't I would like to see GenerateMethod above (and therefore being the default). This would require the removal of this ExtensionOrder attribute and a change of 

ExtensionOrder(**Before** = PredefinedCodeFixProviderNames.GenerateConstructor)]

to

ExtensionOrder(**After** = PredefinedCodeFixProviderNames.GenerateConstructor)]

Is this OK?